cpp/util/etcd.cc

```diff
@@ -668,10 +670,36 @@ bool EtcdClient::Node::HasExpiry() const {
 
 
 EtcdClient::EtcdClient(UrlFetcher* fetcher, const string& host, uint16_t port)
-    : fetcher_(CHECK_NOTNULL(fetcher)), endpoint_(host, port) {
-  CHECK(!endpoint_.first.empty());
-  CHECK_GT(endpoint_.second, 0);
+    : EtcdClient(fetcher, list<HostPortPair>{HostPortPair(host, port)}) {
+}
+
+
+EtcdClient::EtcdClient(UrlFetcher* fetcher, const list<HostPortPair>& etcds)
+    : fetcher_(CHECK_NOTNULL(fetcher)), etcds_(etcds) {
+  CHECK(!etcds_.empty()) << "No etcd hosts provided.";
   VLOG(1) << "EtcdClient: " << this;
+
+  for (const auto& e : etcds_) {
+    CHECK(!e.first.empty()) << "Empty host specified";
+    CHECK_GT(e.second, 0) << "Invalid port specified";
+  }
+}
+
+
+EtcdClient::HostPortPair EtcdClient::endpoint() const {
```

Comment:
That looks a **lot** like `GetEndpoint()`! I don't mind renaming it, but not duplicating. :wink: